### DISCUSSION TITLE
Plugin development: Testing plugin registration via pytester / testdir

### DISCUSSION THREAD

**Discussion by symonk** [pytest-dev/pytest#7994] [Q&A]
Pseudo scenario:

```
Given I am a plugin developer, I have some checks to avoid registering my plugin with the plugin manager.
When I am writing tests with pytester / testdir.
Then there is a simple mechanism to check if my plugin was registered or not.
```

The simplest way to achieve such a goal I can (tell) is through an ugly test, accessing private attrs like so:

```python

from infrastructure import INFRASTRUCTURE_PLUGIN_NAME

def test_when_bypassed_infra_no_plugin_is_registered(testdir) -> None:
    result = testdir.runpytest("--skip-infra")
    plugins_dict = result.reprec._pluginmanager._name2plugin
    assert INFRASTRUCTURE_PLUGIN_NAME not in plugins_dict
```

Is there a cleaner way? I've had a look through the docs and nothing is jumping out.

**Comment by RonnyPfannschmidt** [comment]
there is no convention for this actually, the general way currently is to ensure they are installed and then testing the functionality

**Comment by symonk** [comment]
Thanks @RonnyPfannschmidt i would like to see it being a little easier, as it seems pretty common practice to defer plugins registration under certain scenarios

**Reply by RonnyPfannschmidt** [reply]
installation vs functionality activation are distinct, any installed plugin is active, its common to enable extra plugin components on condition
